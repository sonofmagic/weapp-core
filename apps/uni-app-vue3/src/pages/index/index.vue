<template>
  <view>

  </view>
</template>

<script setup lang="ts">

import { ref } from 'vue'
import { onLoad } from '@dcloudio/uni-app'
// import { io } from 'socket.io-client'
onLoad(() => {
  console.log('onLoad', XMLHttpRequest, WebSocket)
  // const socket = io('http://localhost:3000/events', {
  //   reconnectionDelayMax: 10000,
  //   auth: {
  //     token: '123'
  //   },
  //   query: {
  //     'my-key': 'my-value'
  //   },
  //   transports: ['polling', 'websocket']
  // })
  // socket.on('connect', function () {
  //   console.log('Connected')

  //   socket.emit('events', { test: 'test' })
  //   socket.emit('identity', 0, (response) => console.log('Identity:', response))
  // })
  // socket.on('events', function (data) {
  //   console.log('event', data)
  // })
  // socket.on('exception', function (data) {
  //   console.log('event', data)
  // })
  // socket.on('disconnect', function () {
  //   console.log('Disconnected')
  // })

  // socket.on('connect_error', (error) => {
  //   console.error(error)
  // })
})
</script>

<style></style>
