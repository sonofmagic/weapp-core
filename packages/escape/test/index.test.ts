import { MappingChars2String, MappingChars2StringEntries, SimpleMappingChars2String, SimpleMappingChars2StringEntries, isAsciiNumber, escape } from '@/index'

const testCase = `1234567890-=\b~!@#$%^&*()_+qwertyuiop[]\\QWERTYUIOP{}|asdfghjkl;'ASDFGHJKL:"zxcvbnm,./ZXCVBNM<>?`
describe('index', () => {
  it('map count eq', () => {
    expect(MappingChars2StringEntries.length).toBe(SimpleMappingChars2StringEntries.length)
    expect(Object.keys(MappingChars2String).sort()).toEqual(Object.keys(SimpleMappingChars2String).sort())
  })

  it('isAsciiNumber', () => {
    for (const char of '1234567890') {
      const num = char.codePointAt(0)
      if (num) {
        const res = isAsciiNumber(num)
        // if (!res) {
        //   console.log(char)
        // }
        expect(res).toBe(true)
      }
    }
  })

  it('not AsciiNumber', () => {
    for (const char of "qwertyuiop[]asdfghjkl;'zxcvbnm,./=-") {
      const num = char.codePointAt(0)
      if (num) {
        const res = isAsciiNumber(num)

        expect(res).toBe(false)
      }
    }
  })

  it('escape', () => {
    const res = escape(testCase)
    expect(res).toMatchSnapshot()
  })

  it('escape with map', () => {
    const res = escape(testCase, { map: MappingChars2String })
    expect(res).toMatchSnapshot()
  })
})
