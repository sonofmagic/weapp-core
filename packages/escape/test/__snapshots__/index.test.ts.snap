// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`index > escape 1`] = `"_1234567890-etijhrpknm___aqwertyuiop__wQWERTYUIOP__uasdfghjklfqASDFGHJKLcxzxcvbnmbdsZXCVBNMlgo"`;

exports[`index > escape with map 1`] = `"_1234567890-_e__t__i__at__h__do__p__ca__am__a__pl__qr___plus_qwertyuiop_bl__br__bs_QWERTYUIOP_bal__bar__b_asdfghjkl_se__q_ASDFGHJKL_c__dq_zxcvbnm_co__d__s_ZXCVBNM_l__g__qu_"`;
